### PR TITLE
Homework3: Maxim Sankouski 

### PR DESCRIPTION
**Pull request by MaxSnk**

### CODE REVIEW

File: src/3_7_Chocolate_bar.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+такую площадь.
+Т.к. отломить надо за один раз, то
+одна из сторон отломанной части k должна
+быть равна или n или m. Значит, деление k на
+n или m должно давать целое число."""
+
+# shortest 3th version
+if k < n * m:
+    print(k % n == 0 or k % m == 0)
+
```
Comment (belovmd) [inline src/3_7_Chocolate_bar.py]: it's possible for any k<m*n.

File: src/3_7_Chocolate_bar.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        elif k // n == 0 or k // m == 0:
+            """Если одним разломом искомые дольки уже не
+            отломить, два условия выше становятся
+             направлены на отламывание ненужных"""
+
+            k = n * m - k
+
+    print("Количество разломов: ", razl)
+
+"""Задача не выполняется оптимально. Например,
```
Comment (belovmd) [inline src/3_7_Chocolate_bar.py]: Question is if we can cut them, no matter how. and k is input from user

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework3.md):
- 15
- 16
- 28
- 29
- 2
- 18
- 30
- 4
- 11

File: src/3_7_Chocolate_bar.py (hunk 1 above)
Comment (MaxSnk) [reply src/3_7_Chocolate_bar.py]: Chocolate is a physical object. Breaking - action on it. In the first subtask, we need a single chocolate break. We generally cannot make a break when a piece is equal to or greater than a physical object.

File: src/3_7_Chocolate_bar.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                    razl += 2
+                    k = 0
+        print("Минимальное количество разломов:", razl)
+
+    else:
+        print('Кусок больше возможного')
+
+
+print('\nВторая задача:')
+# сразу отламываем от одной стороны:
```
Comment (belovmd) [inline src/3_7_Chocolate_bar.py:130]: redundant comment (p.28 from review was not addressed)

File: src/3_2_List_practice.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+прошлого списка и напечатайте его.
+5. Скопируйте список и добавьте в него элемент
+'2a' так чтобы в исходном списке этого элемента
+не было."""
+
+
+import copy
+
+list0 = 'abcd'
+list1 = [element1 + element2 for element1 in list0[0:2]
```
Comment (belovmd) [inline src/3_2_List_practice.py:18]: el1 in 'ab' el2 in 'cd'
your code pretty tricky

File: src/3_4_Couples.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+for number in range(len(list1)):
+    list1[number] = int(list1[number])
+
+list1.sort()
+
+list2 = list1.copy()
+list2.reverse()
+
+for element in list1:
+    if element not in list_temp:
```
Comment (belovmd) [inline src/3_4_Couples.py:25]: O(n**2) solution p.4 review not addressed

File: src/3_4_Couples.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+        number_of_element = \
+            len(list1) - list2.index(element) - list1.index(element)
+
+        number_of_couples = \
```
Comment (belovmd) [inline src/3_4_Couples.py:30]: redundant \. code looks pretty difficult.

File: src/3_5_Unic_elements.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+встречаются в списке."""
+
+
+list1 = [1, 2, 2, 3, 7, 5, 6, 4, 1, 5, 6, 'f']
+
+list2 = list1.copy()
+list2.reverse()
+
+for element in list1:
+    if (len(list1) - list2.index(element) - list1.index(element)) == 1:
```
Comment (belovmd) [inline src/3_5_Unic_elements.py:14]: it's task for dict. dct[a] = dct.get(a) +1 i.e. 
you solution O(n**2). p.11 review not addressed.

**Review by belovmd — COMMENTED** [review]
7,5

**Pull request closed without merging**